rust-runtime/aws-smithy-json/src/deserialize/token.rs

```diff
@@ -216,9 +216,22 @@ pub fn expect_timestamp_or_null(
     timestamp_format: Format,
 ) -> Result<Option<DateTime>, Error> {
     Ok(match timestamp_format {
-        Format::EpochSeconds => {
-            expect_number_or_null(token)?.map(|v| DateTime::from_secs_f64(v.to_f64_lossy()))
-        }
+        Format::EpochSeconds => expect_number_or_null(token)?
+            .map(|v| v.to_f64_lossy())
+            .map(|v| {
+                if v.is_infinite() || v.is_nan() {
```

Comment:
I think I would not conflate both cases. There's no harm in being explicit and provide two distinct error messages.